components/core/src/clp/streaming_archive/single_file_archive/writer.cpp

```diff
@@ -0,0 +1,289 @@
+#include "writer.hpp"
+
+#include <cstring>
+#include <filesystem>
+#include <sstream>
+#include <vector>
+
+#include <fmt/core.h>
+#include <msgpack.hpp>
+#include <spdlog.h>
+
+#include "../../Defs.h"
+#include "../../ErrorCode.hpp"
+#include "../../FileReader.hpp"
+#include "../../FileWriter.hpp"
+#include "../../TraceableException.hpp"
+#include "../ArchiveMetadata.hpp"
+#include "../Constants.hpp"
+#include "Defs.hpp"
+
+namespace clp::streaming_archive::single_file_archive {
+
+namespace {
+constexpr size_t cReadBlockSize = 4096;
+
+/**
```

Comment:
I personally feel "packer" might be a better name, since this file isn't a class, but just a collection of helper method.

In addition, I am not sure if this is the best design method. @kirkrodrigues are we using similar packer design in clp-s?

Reply:
Sure, we can use writer for now